articles/quickstart/spa/vanillajs/interactive.md

```diff
@@ -0,0 +1,134 @@
+
+## Add Login to Your Application {{{ data-action=code data-code="app.js#6:12" }}}
+
+The Auth0 SPA SDK gives you tools to quickly implement user authentication in your JavaScript application. Executing `loginWithRedirect()` redirects your users to the Auth0 Universal Login Page, where Auth0 can authenticate them. Upon successful authentication, Auth0 will redirect your users back to your application.
+
+Create a button in your application that calls `loginWithRedirect()` when clicked.
+
+::::checkpoint
+::: checkpoint-default
+Once a login button and the callback handler have been added, verify that you can click the button and are redirected to Auth0 for authentication, and that you do not receive any errors in the console relating to Auth0.
+:::
+
+:::checkpoint-failure
+Sorry about that. Here's a couple things to double check:
+
+- make sure the correct application is selected
+- did you save after entering your URLs?
+- make sure the Auth0 client has been correctly configured with your Auth0 domain and client ID
+
+Still having issues? Check out our [documentation](https://auth0.com/docs) or visit our [community page](https://community.auth0.com) to get more help.
+
+:::
+::::
+
+## Handle the Callback from Auth0 {{{ data-action=code data-code="app.js#14:19" }}}
+
+To process the result from Auth0 when the browser is redirect back to your application, call the `handleRedirectCallback()` function on the Auth0 client.
+
+This should be called somewhere on page load when a callback from Auth0 is detected. One way to do this is to only call `handleRedirectCallback()` when there are `code` and `state` query parameters detected.
+
+If handling the callback was successful, the parameters should be removed from the URL so as to not trigger the callback handler the next time the page loads.
+
+::::checkpoint
+::: checkpoint-default
+Add the logic to handle the callback from Auth0 and click the login button again. Verify that Auth0 successfully redirects back to your application after authentication, and that the query parameters are removed from the URL.
+:::
+
+::: checkpoint-failure
+Sorry about that. Here's a couple things to double check:
+
+- check that the `redirect_uri` option has been configured to your application's URL
+- if you have an `error` query parameter, inspect it to learn the cause of the error
+
+Still having issues? Check out our [documentation](https://auth0.com/docs) or visit our [community page](https://community.auth0.com) to get more help.
+:::
+::::
+
+## Add Logout to Your Application {{{ data-action=code data-code="app.js#21:27" }}}
+
+Now that you can log in to your application, you need [a way to log out](/logout/guides/logout-auth0). The Auth0 client provides a `logout()` method that you can use to log a user out of your app. Executing `logout()` redirects your users to your [Auth0 logout endpoint](/api/authentication?javascript#logout) and then immediately redirects them back to your application.
```

Comment:
Will the logout link go to `https://auth0.com/logout/guides/logout-auth0`? (404)

Reply:
Same for `/api` - not sure if these automatically get the `/docs` prefix

Reply:
Yes - this content was borrowed from [the Angular quickstart](https://auth0.com/docs/quickstart/spa/angular#add-logout-to-your-application) where you can verify that the links work.